libs/core/src/schemas/entities.schemas.ts

```diff
@@ -487,8 +487,11 @@ export const Outbox = z.object({
     events.DiscordMessageCreated.extend({
       event_name: z.literal(EventNames.DiscordMessageCreated),
     }),
+    events.ChainEventCreated.extend({
+      event_name: z.literal(EventNames.ChainEventCreated),
```

Comment:
this might be redundant, the event name is already a field above

Reply:
You asked me to include this to ensure the event matches the payload. I agree it is redundant but I'm not sure how to ensure the payload matches the event name otherwise.

Reply:
do we need the other field then? if we already know the payload must have an event name?

Reply:
Strictly speaking, we could probably use one or the other. That said, I prefer having the `event_name` be a separate column because it:
1. Makes `Outbox` records more closely match the `EventContext` which has a separate `name` and `payload` properties.
2. Facilitates queries on the Outbox table. For example, we `NOTIFY` a channel with the `event_name` - without this column, we would need to JSON parse the `event_payload` inside of the trigger to get the name. This is dangerous because there is no `NOT NULL` constraint specifically on the `event_name` property. The workaround would be creating a CHECK constraint to ensure the payload matches but this adds unnecessary overhead and complexity to the DB and Sequelize models.

I think I may have found a Zod schema that separates the `event_name` from the `event_payload` but still ensures the name and payload match:

```
const BaseOutboxProperties = z.object({
  event_id: z.number().optional(),
  relayed: z.boolean().optional(),
  created_at: z.date().optional(),
  updated_at: z.date().optional(),
});

export const Outbox = z.union([
  z.object({
    event_name: z.literal(EventNames.ThreadCreated),
    event_payload: events.ThreadCreated,
  }).merge(BaseOutboxProperties),
  z.object({
    event_name: z.literal(EventNames.CommentCreated),
    event_payload: events.CommentCreated,
  }).merge(BaseOutboxProperties),
  z.object({
    event_name: z.literal(EventNames.GroupCreated),
    event_payload: events.GroupCreated,
  }).merge(BaseOutboxProperties),
  z.object({
    event_name: z.literal(EventNames.CommunityCreated),
    event_payload: events.CommunityCreated,
  }).merge(BaseOutboxProperties),
  z.object({
    event_name: z.literal(EventNames.SnapshotProposalCreated),
    event_payload: events.SnapshotProposalCreated,
  }).merge(BaseOutboxProperties),
  z.object({
    event_name: z.literal(EventNames.DiscordMessageCreated),
    event_payload: events.DiscordMessageCreated,
  }).merge(BaseOutboxProperties),
  z.object({
    event_name: z.literal(EventNames.ChainEventCreated),
    event_payload: events.ChainEventCreated,
  }).merge(BaseOutboxProperties),
]);
```

This is more verbose but provides both type safety and the benefits of name + payload separation.

Reply:
Yep, that's the idea... I'm also pointing to the extra metadata we might need to include in all events here:
[libs/core/src/schemas/events.schemas.ts](https://github.com/hicommonwealth/commonwealth/pull/7404/files#:~:text=export%20const-,EventMetadata,-%3D%20z.)